akka-cluster/src/test/scala/akka/cluster/ClusterSpec.scala

```diff
@@ -73,9 +73,22 @@ class ClusterSpec extends AkkaSpec(ClusterSpec.config) with ImplicitSender {
     }
 
     "fail fast in a join if invalid chars in host names, e.g. docker host given name" in {
-      val address = Address("akka", "sys", Some("in_valid"), Some(0))
-      intercept[IllegalArgumentException](cluster.join(address))
-      intercept[IllegalArgumentException](cluster.joinSeedNodes(scala.collection.immutable.Seq(address)))
+      val addresses = scala.collection.immutable
+        .Seq(Address("akka", "sys", Some("in_valid"), Some(0)), Address("akka", "sys", Some("invalid._org"), Some(0)))
+
+      addresses.foreach(a => intercept[IllegalArgumentException](cluster.join(a)))
+      intercept[IllegalArgumentException](cluster.joinSeedNodes(addresses))
+    }
+
+    "not fail fast to attempt a join with valid chars in host names" in {
+      val addresses = scala.collection.immutable.Seq(
+        Address("akka", "sys", Some("localhost"), Some(0)),
+        Address("akka", "sys", Some("is_valid.org"), Some(0)),
+        Address("akka", "sys", Some("fu.is_valid.org"), Some(0)),
+        Address("akka", "sys", Some("fu_.is_valid.org"), Some(0)))
```

Comment:
I would have thought this is still invalid because `_` in host part of entry?

Reply:
`_[^.]*$` The trailing `.org` makes it valid.

Reply:
Notably because many of the tickets from docker or related specified that the underscore if valid if part of a fully-qualified domain, yet invalid if not.

Reply:
Aha, I see. 👍